### ISSUE TITLE
WinUI NavigationViewItem can't have set an Icon

### ISSUE THREAD

**Issue by Brouilles** [microsoft/react-native-xaml#123]
I have setup a very simple project. And the NavigationView don't seems to have the WinUI 2.6 design. No rounded corner at hover. And old settings icon. How I can update? Or it is planned?

![image](https://user-images.githubusercontent.com/1573282/133647792-261d1ae2-060b-4f86-99c7-12f861f184eb.png)

Thanks!

**Comment by chrisglein** [comment]
Until fairly recently (version 0.65), the default WinUI version included by React Native for Windows was earlier than 2.6. I believe it was WinUI 2.3 on 0.64? Anyway, with the latest version the new app template should give you WinUI 2.6 styles. What version is your app pulling in?

You can provide some more environment info with these commands:
1. `npx react-native --version`:
2. `npx react-native info`:

There should be some documentation on how to easily update your WinUI version (outside of adjusting the nuget version directly in Visual Studio) - I'm trying to track that down.

**Comment by chrisglein** [comment]
Here's that documentation: https://microsoft.github.io/react-native-windows/docs/customizing-sdk-versions

**Comment by Brouilles** [comment]
Thanks for your answer.

npx react-native --version:
```
6.0.0
```

npx react-native info:
```
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i5-9300HF CPU @ 2.40GHz
    Memory: 2.55 GB / 7.92 GB
  Binaries:
    Node: 14.17.6 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 7.21.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.16299.0, 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version     2020.3.0.0 AI-203.7717.56.2031.7678000
    Visual Studio: 16.11.31702.278 (Visual Studio Community�2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.65.0 => 0.65.0
    react-native-windows: 0.65.2 => 0.65.2
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by Brouilles** [comment]
And inside the Visual Studio project and `node_modules\react-native-windows\Microsoft.ReactNative\packages.config` I have WinUI 2.6.

packages.config
```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="boost" version="1.72.0.0" targetFramework="native" />
  <package id="Microsoft.Build.Tasks.Git" version="1.0.0" targetFramework="native" developmentDependency="true" />
  <package id="Microsoft.SourceLink.Common" version="1.0.0" targetFramework="native" developmentDependency="true" />
  <package id="Microsoft.SourceLink.GitHub" version="1.0.0" targetFramework="native" developmentDependency="true" />
  <package id="Microsoft.UI.Xaml" version="2.6.0" targetFramework="native" />
  <package id="Microsoft.Windows.CppWinRT" version="2.0.210312.4" targetFramework="native" />
  <package id="Microsoft.WinUI" version="3.0.0-preview4.210210.4" targetFramework="native" />
  <package id="ReactNative.Hermes.Windows" version="0.8.0-ms.0" targetFramework="native" />
  <!-- package id="ReactNative.V8Jsi.Windows.UWP" version="0.65.2" targetFramework="native" / -->
</packages>
```

So I don't understand why I don't have the good design.

**Comment by asklar** [comment]
What's your JSX like? Make sure you are using NavigationView from WinUI, not the plain one (as that uses system xaml).
i.e. `<WinUI.NavigationView ...>`

**Comment by chrisglein** [comment]
> What's your JSX like? Make sure you are using NavigationView from WinUI, not the plain one (as that uses system xaml).
> i.e. `<WinUI.NavigationView ...>`

Interesting. I didn't realize that the styling for NavView wasn't just a style change via generic.xaml, but required that you use the different namespaced control.

If you look at the [example in react-native-gallery](https://github.com/microsoft/react-native-gallery/blob/main/src/examples/XamlExamplePage.tsx#L157) it does _not_ include the `WinUI.` namespace prefix. I'm not seeing the rounded NavView items in the current gallery version (@chiaramooney FYI) even though it does pull in the [XamlControlResources in App.xaml](https://github.com/microsoft/react-native-gallery/blob/main/windows/rngallery/App.xaml#L8).

**Comment by asklar** [comment]
yeah that's because we added the react-native-xaml controls to gallery before I added the "use winui 2.x" feature to the react-native-xaml codegen :)

**Comment by Brouilles** [comment]
My bad. This is it. Before I use the plain `NavigationView`. Not the `WinUI.NavigationView`. So it is why.
Thanks for your time and patient!

Maybe you can help, I don't how to set an Icon now. Seems different than with the "legacy" NavigationViewItem. I have try this but don't work.

```
<WinUI.NavigationViewItem content={{string: 'Tableau de bord'}} icon={{string: 'Home'}}>
        </WinUI.NavigationViewItem>
```

Nior

```
<WinUI.NavigationViewItem content={{string: 'Clients'}}>
          <FontIcon glyph="&#xE790;" />
        </WinUI.NavigationViewItem>
```

**Comment by chrisglein** [comment]
> Maybe you can help, I don't how to set an Icon now. Seems different than with the "legacy" NavigationViewItem. I have try this but don't work.
> 
> ```js
> <WinUI.NavigationViewItem content={{string: 'Tableau de bord'}} icon={{string: 'Home'}}>
>         </WinUI.NavigationViewItem>
> ```

The `NavigationViewItem.Icon` is supposed to take a value from the [`Symbol`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.symbol?view=winrt-20348) enumeration. Is the conversion from string "Home" to Symbol working?

Can you inspect your UI with the [Live Visual Tree](https://docs.microsoft.com/en-us/visualstudio/xaml-tools/inspect-xaml-properties-while-debugging?view=vs-2019) and see if you're getting the NavigationViewItems you expect?

**Comment by asklar** [comment]
@Brouilles the fact that it crashes when setting an icon is a bug, fix is in #124
